lm_eval/api/task.py

```diff
@@ -785,3 +785,42 @@ def get_logging_info(self):
         return {
             "prompt_name": None,
         }
+
+
+class ChainOfThoughtPromptingTask(PromptSourceTask):
+
+    def has_chain_of_thought_docs(self):
+        """Whether the task has a chain-of-thought set"""
+        return True
+
+    def chain_of_thought_docs(self) -> datasets.Dataset:
+        """
+        Returns:
+            A subset of samples with chain-of-thought examples..
+        """
+        raise NotImplementedError("Needs a set of examples")
+
+    def fewshot_docs(self) -> datasets.Dataset:
+        """Returns the `dataset` split that the few-shot examples should be sample
```

Comment:
I think the docstring needs to be updated w.r.t. CoT; I was thinking something along the lines of "Returns the `dataset` split that the few-shot examples should be sampled from. This prioritizes the `chain_of_thought_docs` as the few-shot example. If the task does not have that, then it falls back to the `fewshot_docs` logic from the parent class."